src/pages/home/sidebar/OptionRowTitle/index.js

```diff
@@ -0,0 +1,120 @@
+        };
+        this.setOptionChildRef = index => (ref) => {
+            this.cRefs[index] = ref;
+        };
+        this.state = {
+            isEllipsisActive: false,
+            containerLayout: null,
+        };
+        this.setContainerLayout = this.setContainerLayout.bind(this);
+        this.getTooltipShiftX = this.getTooltipShiftX.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({
+            isEllipsisActive: this.ref && hasEllipsis(this.ref),
+        });
+    }
+
+    setContainerLayout({nativeEvent}) {
+        this.setState({
+            containerLayout: nativeEvent.layout,
+        });
+    }
+
+    getTooltipShiftX(index) {
+        const {containerLayout} = this.state;
+
+        // only shift when containerLayout or Refs to text node is available .
+        if (!containerLayout || !this.cRefs[index]) {
+            return;
+        }
+        const {width: cWidth, left: cLeft} = containerLayout;
+
+        // we have to return the value as Number so we can't use `measureWindow` which takes a callback
+        const {width: tWidth, left: tLeft} = this.cRefs[index].getBoundingClientRect();
+        const toolX = (tWidth / 2) + tLeft;
+        const cRight = cWidth + cLeft;
+        const tRight = tWidth + tLeft;
+        const newToolX = tLeft + ((cRight - tLeft) / 2);
+
+        // when text right end is beyond the Container Right end
+        return tRight > cRight ? -(toolX - newToolX) : 0;
+    }
+
+
+    render() {
+        const {
+            option, style, tooltipEnabled, numberOfLines, tooltipContainerStyle,
+        } = this.props;
+        const {isEllipsisActive} = this.state;
+
+        if (tooltipEnabled) {
+            return (
+
+                // Tokenization of string only support 1 numberofLines on Web
+                <Text
+                    style={[style, styles.optionDisplayNameTooltipWrapper]}
+                    onLayout={this.setContainerLayout}
+                    numberOfLines={1}
+                    ref={this.setContainerRef}
+                >
+                    {_.map(option.participantsList, (participant, index) => {
+                        const setChildRef = this.setOptionChildRef(index);
+                        return (
+                            <Fragment key={index}>
+                                <Tooltip
+                                    key={index}
+                                    text={participant.login}
+                                    containerStyle={tooltipContainerStyle}
+                                    shiftHorizontal={() => this.getTooltipShiftX(index)}
+                                >
+                                    <Text ref={setChildRef}>
+                                        {participant.displayName}
+                                    </Text>
+                                </Tooltip>
+                                {index < option.participantsList.length - 1 ? <Text>,&nbsp;</Text> : null}
```

Comment:
Use `&&` instead of ternary.

Reply:
It was mentioned somewhere in the code that We should return null as it was causing the issue on IOS.

Reply:
It's OK to return `null`. What you want to avoid is doing a truthy/falsey check on a string. For example:

```
const string = '';
// good
{!_.isEmpty(string) && <View>{string}</View>}

// bad
{string && <View>{string}</View>}
```

The second one will render an empty `<Text>` component on iOS when the string is empty, which is the issue being avoided.

Reply:
Ah. Sorry missed it. It is mentioned on the styleGuide.